libnavigation-core/src/main/java/com/mapbox/navigation/core/internal/LegacyMapboxNavigationInstanceHolder.kt

```diff
@@ -0,0 +1,46 @@
+package com.mapbox.navigation.core.internal
+
+import androidx.annotation.UiThread
+import androidx.annotation.VisibleForTesting
+import com.mapbox.navigation.core.MapboxNavigation
+import java.util.concurrent.CopyOnWriteArraySet
+
+fun interface MapboxNavigationCreateObserver {
+
+    fun onCreated(mapboxNavigation: MapboxNavigation)
+}
+
+@Deprecated("Used to keep track of MapboxNavigation instances created via deprecated methods")
```

Comment:
This is weird, why are we adding a new `@Deprecated` `object`? 🤔

Reply:
I have to register RoutesObserver and RouteProgressObserver. And there are multiple ways of creating `MapboxNavigation` :
1) constructor
2) MapboxNavigationProvider  (deprecated)
M3) apboxNavigationApp
The "correct" way of observing `MapboxNavigation` is using `MapboxNavigationApp` but the other ways are widely used and I want the feature to work for all the users. So `LegacyMapboxNavigationInstanceHolder` tracks all `MapboxNavigation` instances. Not sure if the naming is very good but I wanted to emphasize by this that the `MapboxNavigationApp` is the appropriate way to do it and `LegacyMapboxNavigationInstanceHolder` is just for us to support other creation methods.

Reply:
`LegacyMapboxNavigationInstanceHolder` is an internal object. User aren't supposed to use it anyway, are they?

Reply:
They aren't. It's just a note for us. I can just add a comment or sth instead of deprecating it.
But actually I'm thinking that it should not be deprecated at all. We still expose a valid way of creating `MapboxNavigation` via constructor and some customers use it. So it's not really "legacy", it's more like "everything including legacy". I just don't like it that we have `MapboxNavigationApp` and this new way of observing. But since the new one is internal, it's not that big a deal.

Reply:
`LegacyMapboxNavigationInstanceHolder` is definitely not an improvement to `MapboxNavigationApp` though, in my opinion.

`MapboxNavigationCreateObserver` is also redundant.. It is the same as `MapboxNavigationObserver.onAttached`.

Reply:
I's not, see https://github.com/mapbox/mapbox-navigation-android/pull/6547#discussion_r1054079256.

Reply:
> have to register RoutesObserver

This is not a reason to require an automatic link. It can be done after MapboxNavigation is constructed